### ISSUE TITLE
[vLLM] Encountered an out-of-memory (OOM) error when running the Mistral model

### ISSUE THREAD

**Issue by devisettymahidhar608** [tenstorrent/tt-xla#5144]
While running vLLM with the Mistral-Small-3.1 multimodal model on TT-XLA, we are encountering the following error:

```
%304 = ttnn.matmul(...)        -> 1x4x12100x12100   loc("add.700_decomp_matmul")   # Q @ Kᵀ
%305 = ttnn.add(%304, %2)      -> 1x4x12100x12100   loc("add.700_decomp_add")      # + attn_mask
%306 = ttnn.eq(%305, %9)       -> 1x4x12100x12100   loc("compare.727")             # == -inf
%307 = ttnn.prod(%306, dim=3)  -> 1x4x12100x1       loc("reduce.739")   <-- OOM thrown here
```

The OOM is thrown by ttnn.prod reducing dim 3 of a 1x4x12100x12100 bf16 attention tensor (12100 = 110×110 vision patches). That prod is part of the "safe-softmax" decomposition of a single nn.functional.scaled_dot_product_attention call in Pixtral's vision encoder. ttnn::prod reduces a dim by permuting it first, which materializes a full-size copy of the ~1 GB score tensor (and more transient buffers), exhausting DRAM.

Repro:
```
git clone https://github.com/tenstorrent/tt-xla.git
cd tt-xla
git submodule update --init --recursive
git checkout devisettym/mistral-vllm-integration

then try to run the following command:
python -m pytest -svv "tests/integrations/vllm_plugin/generative/test_tensor_parallel_generation.py::test_tensor_parallel_generation_mistral_small"


```
